presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -469,4 +470,25 @@ public String toString()
                 .add("estimatedSize", estimatedSize)
                 .toString();
     }
+
+    public Iterator<Page> getPages(int[] outputChannels)
+    {
+        return new Iterator<Page>() {
+            int currentPosition = 0;
+            PageBuilder pageBuilder = new PageBuilder(types);
+            @Override
+            public boolean hasNext()
+            {
+                pageBuilder.reset();
+                currentPosition = buildPage(currentPosition, outputChannels, pageBuilder);
+                return !pageBuilder.isEmpty();
```

Comment:
That getPages function will return pages in the input layout. I think sorting only happens on sort of the "index" of the rows (valueAddresses) so unless you build new pages according to that ordering, you won't get sorted result.